tooling/hcpctl/cmd/must-gather/options_clean.go

```diff
@@ -0,0 +1,141 @@
+	if opts.BaseOptions == nil {
+		return fmt.Errorf("base options cannot be nil")
+	}
+	if err := base.BindBaseOptions(opts.BaseOptions, cmd); err != nil {
+		return fmt.Errorf("failed to bind base options: %w", err)
+	}
+	cmd.Flags().StringVar(&opts.PathToClean, "path-to-clean", opts.PathToClean, "Path to clean")
+	cmd.Flags().StringVar(&opts.ServiceConfigPath, "service-config-path", opts.ServiceConfigPath, "Path to ARO-HCP Service Configuration file (not must-gather-clean config)")
+	cmd.Flags().StringVar(&opts.MustGatherCleanBinary, "must-gather-clean-binary", opts.MustGatherCleanBinary, "Path to must-gather-clean binary")
+	cmd.Flags().StringVar(&opts.CleanedOutputPath, "cleaned-output-path", opts.CleanedOutputPath, "Path to cleaned output")
+	cmd.Flags().StringVar(&opts.CleanConfigPath, "clean-config-path", opts.CleanConfigPath, "Path to must-gather-clean config, will be extended with ARO-HCP Service Configuration literals")
+
+	if err := cmd.MarkFlagDirname("path-to-clean"); err != nil {
+		return fmt.Errorf("failed to mark flag %q as a file: %w", "path-to-clean", err)
+	}
+	if err := cmd.MarkFlagRequired("path-to-clean"); err != nil {
+		return fmt.Errorf("failed to mark flag %q as a required: %w", "config-file-path", err)
+	}
+	if err := cmd.MarkFlagDirname("service-config-path"); err != nil {
+		return fmt.Errorf("failed to mark flag %q as a file: %w", "service-config-path", err)
+	}
+	if err := cmd.MarkFlagRequired("service-config-path"); err != nil {
+		return fmt.Errorf("failed to mark flag %q as a required: %w", "service-config-path", err)
+	}
+	if err := cmd.MarkFlagFilename("must-gather-clean-binary"); err != nil {
+		return fmt.Errorf("failed to mark flag %q as a file: %w", "must-gather-clean-binary", err)
+	}
+	if err := cmd.MarkFlagRequired("must-gather-clean-binary"); err != nil {
+		return fmt.Errorf("failed to mark flag %q as a required: %w", "must-gather-clean-binary", err)
+	}
+	if err := cmd.MarkFlagDirname("cleaned-output-path"); err != nil {
+		return fmt.Errorf("failed to mark flag %q as a directory: %w", "cleaned-output-path", err)
+	}
+	return nil
+}
+
+type ValidatedCleanOptions struct {
+	*RawCleanOptions
+}
+
+type CleanOptions struct {
+	*ValidatedCleanOptions
+	WorkingDir string
+}
+
+func (opts *RawCleanOptions) Validate(ctx context.Context) (*ValidatedCleanOptions, error) {
+	if opts.PathToClean == "" {
+		return nil, fmt.Errorf("path-to-clean is required")
+	}
+	if _, err := os.Stat(opts.PathToClean); os.IsNotExist(err) {
+		return nil, fmt.Errorf("path-to-clean does not exist")
+	}
+	if opts.ServiceConfigPath == "" {
+		return nil, fmt.Errorf("config-file-path is required")
+	}
+	if _, err := os.Stat(opts.ServiceConfigPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("config-file-path does not exist")
+	}
+	if opts.MustGatherCleanBinary == "" {
+		return nil, fmt.Errorf("must-gather-clean-binary is required")
+	}
+	if _, err := os.Stat(opts.MustGatherCleanBinary); os.IsNotExist(err) {
```

Comment:
Don't bother with these kinds of checks - not only are you checking a subset of the requirements for this path (what if it doesn't have the executable bit set?) but the check is entirely superfluous, as the usage of the binary with `exec.Cmd` will throw an error for *all* cases that are incorrect, and your code already handles that error. Giving the user prior warning about some subset of the errors a few milliseconds before the real error is not necessary.